### ISSUE TITLE
Bug in 'maxInDirection'.

### ISSUE THREAD

**Issue by Lemmih** [noinia/hgeometry#68]
```haskell
convexP :: ConvexPolygon () Rational
convexP = ConvexPolygon $ fromPoints $ map ext
  [Point2 4 3
  ,Point2 5 4
  ,Point2 6 5
  ,Point2 3 5
  ,Point2 1 4
  ]
direction :: Vector 2 Rational
direction = Vector2 (-1) (-1)
```

```haskell
λ> maxInDirection direction convexP
*** Exception: isUpwards: no edge endpoint
CallStack (from HasCallStack):
  error, called at src/Data/Geometry/Polygon/Convex.hs:175:37 in hgeometry-0.11.0.0-inplace:Data.Geometry.Polygon.Convex
```

Found this after adding a quickcheck test for `extremes`.

**Comment by Lemmih** [comment]
Even simpler example:
```haskell
h3 :: ConvexPolygon () Rational
h3 = ConvexPolygon $ fromPoints $ map ext
  [Point2 0 0
  ,Point2 1 0
  ,Point2 1 1
  ]
```

```haskell
λ> maxInDirection (Vector 1 1) h3
*** Exception: isUpwards: no edge endpoint
```

**Comment by Lemmih** [comment]
Fixed in #69.
